src/mat73.c

```diff
@@ -2547,7 +2547,7 @@ Mat_Create73(const char *matname, const char *hdr_str)
     H5Fclose(fid);
     H5Pclose(plist_id);
 
-#if defined(_WIN32) && defined(_MSC_VER) && H5_VERSION_GE(1, 11, 6)
+#if defined(_WIN32) && H5_VERSION_GE(1, 10, 6)
```

Comment:
No, just checked again, HDF5 1.10.6 is the version that changed that behavior. While all HDF5 versions only accept 8bit strings for filenames, 1.10.5 still used the old behavior of directly passing those to the C API functions (hence forcing the usage of the local codepage), while 1.10.6 was the first version HDF5 version that changed behavior on Windows to assume all 8bit strings are UTF-8 and convert them to wide filenames, and use the wide filenames on Windows. So the version check is correct.

See also: https://github.com/HDFGroup/hdf5/compare/hdf5-1_10_5...hdf5-1_10_6 (look at the file `H5system.c` to see the main part of that specific change)

(Btw. I don't think there ever was a 1.11.6 release, since HDF5 uses even minor versions for stable releases.)

Reply:
The odd minor number refer to development version that later get renamed as even minor number.

I cannot confirm that https://github.com/HDFGroup/hdf5/commit/750b5c293076b6a446088fa3020e4e0787d489d7 is available in hdf5-1_10_6, but only in [vfd_swmr_beta_1](https://github.com/HDFGroup/hdf5/releases/tag/vfd_swmr_beta_1) [vfd_swmr_alpha_2](https://github.com/HDFGroup/hdf5/releases/tag/vfd_swmr_alpha_2) [vfd_swmr_alpha_1](https://github.com/HDFGroup/hdf5/releases/tag/vfd_swmr_alpha_1) [snapshot-1.14](https://github.com/HDFGroup/hdf5/releases/tag/snapshot-1.14) [snapshot-1.12](https://github.com/HDFGroup/hdf5/releases/tag/snapshot-1.12) [snapshot](https://github.com/HDFGroup/hdf5/releases/tag/snapshot) [hdf5_1_12-initial](https://github.com/HDFGroup/hdf5/releases/tag/hdf5_1_12-initial) [hdf5-1_14_3](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_14_3) [hdf5-1_14_3-rc1](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_14_3-rc1) [hdf5-1_14_2](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_14_2) [hdf5-1_14_1](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_14_1) [hdf5-1_14_1-2](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_14_1-2) [hdf5-1_14_0](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_14_0) [hdf5-1_13_3](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_13_3) [hdf5-1_13_2](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_13_2) [hdf5-1_13_1](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_13_1) [hdf5-1_13_0](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_13_0) [hdf5-1_13_0-rc6](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_13_0-rc6) [hdf5-1_13_0-rc5](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_13_0-rc5) [hdf5-1_13_0-rc4](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_13_0-rc4) [hdf5-1_13_0-rc3](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_13_0-rc3) [hdf5-1_13_0-rc2](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_13_0-rc2) [hdf5-1_13_0-rc1](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_13_0-rc1) [hdf5-1_12_3](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_12_3) [hdf5-1_12_2](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_12_2) [hdf5-1_12_2-3-rc1](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_12_2-3-rc1) [hdf5-1_12_1](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_12_1) [hdf5-1.12.1-rc1](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1.12.1-rc1) [hdf5-1_12_0](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_12_0) [hdf5-1_12_0-initial](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_12_0-initial) [hdf5-1_12_0-alpha1](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_12_0-alpha1) [hdf5-1_12_0-alpha0](https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_12_0-alpha0) [1.14.1](https://github.com/HDFGroup/hdf5/releases/tag/1.14.1)

It's also not mention in the https://github.com/HDFGroup/hdf5/blob/hdf5-1_10_6/release_docs/RELEASE.txt

Reply:
Thanks for the feeback and clarification. Your personal patch trouble after updating from 1.10.5 and the potential cherry-pick error convinced me finally. (I am only following the HDF5 news and release notes).

My personal request would be to have this as a separate commit/PR, such that I can identifiy it easier. Thanks again.

Reply:
Ok, will split this up